src/jiraUpdateTicket.gs

```diff
@@ -139,16 +147,38 @@ function packageRowForUpdate(allJiraFields, headerRow, dataRow) {
     if (value != null) {
       value = formatFieldValueForJira(fieldDefinition, value);
       if (headerId.toLowerCase() != keyFieldName) {
-        result.fields[headerId] = value;
-
+        // is this a field which needs to be put in fields of an update section
+        if (isIssueScreenField(headerId)) {
+          result.update[headerId] = prepareUpdateField(headerId,value);
+        } else {
+          result.fields[headerId] = value;
+        }
       } else {
         if (value.length > 0) {
           result.key = value;
         }
       }
     }
   }
+  if (Object.keys(result.update).length ==0) {
+    delete(result.update);
+  }
   return result;
+
+  function isIssueScreenField(headerId) {
+    return headerId == "components" || headerId == "fixVersions";
```

Comment:
Its hardcoded with the fields which should use this "update" format. There are a number of places across the code base where the capabilities of the fields are set - I think this may be an opportunity to put that in one place. I will raise that in the PR for refactoring for discussion